algonaut_transaction/src/account.rs

```diff
@@ -94,7 +94,7 @@ impl Account {
     ) -> Result<SignedTransaction, AlgorandError> {
         let transaction_bytes = &transaction.bytes_to_sign()?;
         let signature = self.sign(&transaction_bytes);
-        let id = BASE32_NOPAD.encode(&ChecksumAlg::digest(&signature.0));
+        let id = BASE32_NOPAD.encode(&ChecksumAlg::digest(&transaction.bytes_to_sign()?));
```

Comment:
Can we use `transaction_bytes` instead of building the bytes again?

Reply:
Yes! didn't see that.